kratos/processes/calculate_nodal_area_process.cpp

```diff
@@ -0,0 +1,160 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Riccardo Rossi
+//  Collaborators:   Vicente Mataix Ferrandiz
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "utilities/geometry_utilities.h"
+#include "processes/calculate_nodal_area_process.h"
+
+namespace Kratos
+{
+
+template<bool THistorical>
+void CalculateNodalAreaProcess<THistorical>::Execute()
+{
+    KRATOS_TRY
+
+    // Check if variables are available
+    KRATOS_ERROR_IF(mrModelPart.Nodes().size() == 0) << "No nodes in the model part" << std::endl;
+    if (THistorical) {
+        KRATOS_ERROR_IF_NOT(mrModelPart.NodesBegin()->SolutionStepsDataHas( NODAL_AREA )) << "Variable NODAL_AREA not in the model part!" << std::endl;
+    }
+
+    // Set to zero the nodal area
+    const auto it_node_begin = mrModelPart.NodesBegin();
+    #pragma omp parallel for
+    for(int i=0; i<static_cast<int>(mrModelPart.Nodes().size()); ++i) {
+        auto it_node = it_node_begin + i;
+        SetInitialValue(it_node);
+    }
+
+    const auto& it_element_begin = mrModelPart.ElementsBegin();
+    const auto& r_first_element_geometry = it_element_begin->GetGeometry();
+    const std::size_t local_space_dimension = r_first_element_geometry.LocalSpaceDimension();
+    const std::size_t number_of_nodes = r_first_element_geometry.PointsNumber();
```

Comment:
do you really want to hard-code this?
What if there are e.g. quads and tris in a modelpart?

Reply:
Ok, I will change